"""Initial migration

Revision ID: fe11d1a9d91b
Revises: 9adffa2cc01e
Create Date: 2024-03-11 21:41:47.211075

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'fe11d1a9d91b'
down_revision = '9adffa2cc01e'
branch_labels = None
depends_on = None


def upgrade():
    try:
        # Check if the 'directory' table exists before attempting to create it
        conn = op.get_bind()
        if not conn.dialect.has_table(conn, 'directory'):
            op.create_table('directory',
                sa.Column('id', sa.Integer(), nullable=False),
                sa.Column('name', sa.String(length=100), nullable=False),
                sa.Column('user_id', sa.Integer(), nullable=False),
                sa.PrimaryKeyConstraint('id'),
                sa.ForeignKeyConstraint(['user_id'], ['user.id'], ),
            )
        else:
            print("The 'directory' table already exists.")
    except Exception as e:
        print(f"An error occurred while creating the 'directory' table: {e}")

def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('file', schema=None) as batch_op:
        batch_op.add_column(sa.Column('size', sa.INTEGER(), nullable=True))
        batch_op.drop_constraint(None, type_='foreignkey')
        batch_op.drop_column('directory_id')

    op.drop_table('directory')
    # ### end Alembic commands ###
